packages/devtools_app/lib/src/memory/flutter/memory_heap_tree_view.dart

```diff
@@ -0,0 +1,848 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:intl/intl.dart';
+
+import '../../flutter/auto_dispose_mixin.dart';
+import '../../flutter/table.dart';
+import '../../flutter/theme.dart';
+import '../../table_data.dart';
+import '../../ui/flutter/label.dart';
+import '../../ui/material_icons.dart';
+import '../../utils.dart';
+
+import 'memory_controller.dart';
+import 'memory_filter.dart';
+import 'memory_graph_model.dart';
+import 'memory_heatmap.dart';
+import 'memory_snapshot_models.dart';
+
+class HeapTree extends StatefulWidget {
+  const HeapTree(
+    this.controller,
+  );
+
+  final MemoryController controller;
+
+  @override
+  HeapTreeViewState createState() => HeapTreeViewState();
+}
+
+class HeapTreeViewState extends State<HeapTree> with AutoDisposeMixin {
+  @visibleForTesting
+  static const snapshotButtonKey = Key('Snapshot Button');
+  @visibleForTesting
+  static const groupByClassButtonKey = Key('Group By Class Button');
+  @visibleForTesting
+  static const groupByLibraryButtonKey = Key('Group By Library Button');
+  @visibleForTesting
+  static const collapseAllButtonKey = Key('Collapse All Button');
+  @visibleForTesting
+  static const expandAllButtonKey = Key('Expand All Button');
+  @visibleForTesting
+  static const filterButtonKey = Key('Snapshot Filter');
+  @visibleForTesting
+  static const searchButtonKey = Key('Snapshot Search');
+  @visibleForTesting
+  static const settingsButtonKey = Key('Snapshot Settings');
+
+  MemoryController controller;
+
+  Widget snapshotDisplay;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    controller = Provider.of<MemoryController>(context);
+
+    cancel();
+
+    addAutoDisposeListener(controller.selectedSnapshotNotifier, () {
```

Comment:
this autoDisposeListener calls are listening for some state on the controller to trigger some other state on the controller. 
Instead of doing that, add the listeners directly to the controller and instead have the view just listen for when state changes as a result of computeAllLibraries.